src/common/store/pages/token-dashboard/slice.ts

```diff
@@ -54,30 +54,52 @@ const slice = createSlice({
     inputSendData: (
       state,
       {
-        payload: { amount, wallet }
-      }: PayloadAction<{ amount: StringWei; wallet: WalletAddress }>
+        payload: { amount, wallet, chain }
+      }: PayloadAction<{
+        amount: StringWei
+        wallet: WalletAddress
+        chain: Chain
+      }>
     ) => {
       const newState: ModalState = {
         stage: 'SEND' as const,
         flowState: {
           stage: 'AWAITING_CONFIRMATION',
           amount,
-          recipientWallet: wallet
+          recipientWallet: wallet,
+          chain
         }
       }
       state.modalState = newState
     },
+    transferingEthAudioToSolWAudio: state => {
+      if (
+        state.modalState?.stage !== 'SEND' ||
+        state.modalState.flowState.stage !== 'SENDING'
+      )
+        return
+
+      state.modalState.flowState = {
+        stage: 'AWAITING_CONVERTING_ETH_AUDIO_TO_SOL',
+        recipientWallet: state.modalState.flowState.recipientWallet,
+        amount: state.modalState.flowState.amount,
+        chain: state.modalState.flowState.chain
```

Comment:
since this is using immer, do we need to set all these manually? I think we could just do `state.modalState.flowState.stage = ...` and move on?

Reply:
There's the typescript error because it doesn't know the rest of the shape of flowState here. 
`Type '"SENDING"' is not assignable to type '"AWAITING_CONFIRMATION" | "AWAITING_CONVERTING_ETH_AUDIO_TO_SOL"'.ts(2322)`